src/events/mod.rs

```diff
@@ -833,6 +846,49 @@ impl<'a> arbitrary::Arbitrary<'a> for BytesCData<'a> {
     }
 }
 
+/// Iterator over `CDATA` sections in a string.
+///
+/// This iterator is created by the [`BytesCData::escaped`] method.
+///
+/// If a string contains `]]>`, it needs to be split into multiple `CDATA`
+/// sections, splitting the `]]` and `>` characters. This iterator yields
+/// each of those sections.
+pub struct CDataIterator<'a> {
+    content: &'a str,
```

Comment:
`MatchIndices` already contains something like that, but unfortunately does not give access to it. It could be possible to implement this iterator using memchr, but the current implementation is also fine.